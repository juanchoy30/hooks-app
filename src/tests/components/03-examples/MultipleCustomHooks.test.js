import React from 'react';
import { shallow } from 'enzyme';
import { MultipleCustomHooks } from '../../../components/03-examples/MultipleCustomHooks';
import { useFetch } from '../../../hooks/useFetch';
import { useCounter } from '../../../hooks/useCounter';

jest.mock('../../../hooks/useFetch');
jest.mock('../../../hooks/useCounter');


describe('<MultipleCustomHooks /> tests', () => {

    useCounter.mockReturnValue({
        counter: 10,
        increment: () => {}
    });

    test('should render correctly', () => {

        useFetch.mockReturnValue({
            data: null,
            loading: true,
            error: null
        });
        
        const wrapper = shallow( <MultipleCustomHooks /> );
        expect( wrapper ).toMatchSnapshot();

    });

    test('should show the information', () => {
        
        useFetch.mockReturnValue({
            data: [{
                author: 'Juan Miguel',
                quote: 'Hello World'
            }],
            loading: false,
            error: null
        });
        
        const wrapper = shallow( <MultipleCustomHooks /> );

        expect( wrapper.find('.alert').exists() ).toBe( false );
        expect( wrapper.find('.mb-2').text().trim() ).toBe( 'Hello World' );
        expect( wrapper.find('footer').text().trim() ).toBe( 'Juan Miguel' );
    });
    
        
})
