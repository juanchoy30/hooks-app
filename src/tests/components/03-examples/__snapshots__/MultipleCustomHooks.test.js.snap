// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MultipleCustomHooks /> tests should render correctly 1`] = `
<div>
  <h1>
    BreakingBad Quotes!!
  </h1>
  <hr />
  <div
    className="alert alert-info text-center"
  >
    Loading...
  </div>
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Next Quote
  </button>
</div>
`;
